compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/checkers/call/FirQualifiedSupertypeExtendedByOtherSupertypeChecker.kt

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2010-2020 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.fir.analysis.checkers.call
+
+import org.jetbrains.kotlin.fir.FirSourceElement
+import org.jetbrains.kotlin.fir.analysis.checkers.context.CheckerContext
+import org.jetbrains.kotlin.fir.analysis.checkers.isSupertypeOf
+import org.jetbrains.kotlin.fir.analysis.checkers.toRegularClass
+import org.jetbrains.kotlin.fir.analysis.diagnostics.DiagnosticReporter
+import org.jetbrains.kotlin.fir.analysis.diagnostics.FirErrors
+import org.jetbrains.kotlin.fir.declarations.FirAnonymousObject
+import org.jetbrains.kotlin.fir.declarations.FirClass
+import org.jetbrains.kotlin.fir.declarations.FirRegularClass
+import org.jetbrains.kotlin.fir.expressions.FirQualifiedAccessExpression
+import org.jetbrains.kotlin.fir.references.FirSuperReference
+import org.jetbrains.kotlin.fir.resolve.transformers.firClassLike
+import org.jetbrains.kotlin.utils.addToStdlib.safeAs
+
+object FirQualifiedSupertypeExtendedByOtherSupertypeChecker : FirQualifiedAccessChecker() {
+    override fun check(functionCall: FirQualifiedAccessExpression, context: CheckerContext, reporter: DiagnosticReporter) {
+        // require to be called over a super reference
+        val superReference = functionCall.calleeReference.safeAs<FirSuperReference>()
+            ?: return
+
+        val explicitType = superReference.superTypeRef.toRegularClass(context.session)
+            ?: return
+
+        val surroundingType = context.findClosestClass()
+            ?: return
+
+        // how many supertypes of the `surroundingType`
+        // have `explicitType` as their supertype
+        var count = 0
+
+        for (it in surroundingType.superTypeRefs) {
+            val that = it.firClassLike(context.session).safeAs<FirClass<*>>()
```

Comment:
There can be a typealias in llist of supertype ref that is not a `FirClass`

Consider following example:
```
interface IBase<T> {
    fun foo() {}
    fun bar() {}
}

typealias AliasedIBase = IBase<String>

interface IDerived<T> : IBase<T> {
    override fun foo() {}
    fun qux() {}
}

class Test : IDerived<String>, AliasedIBase {
    fun test() {
        super<IBase>.foo()
        super<IBase>.bar()
        super<AliasedIBase>.foo() // QUALIFIED_SUPERTYPE_EXTENDED_BY_OTHER_SUPERTYPE
        super<AliasedIBase>.bar() // QUALIFIED_SUPERTYPE_EXTENDED_BY_OTHER_SUPERTYPE
    }
}
```

Reply:
You're right, thanks. I'll fix my code

Reply:
I consulted play.kotlinlang.org and it sais that `super<IBase>.bar()` and `super<IBase>.foo()` should also throw `QUALIFIED_SUPERTYPE_EXTENDED_BY_OTHER_SUPERTYPE` so that's the way I made it in my code.